src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Analyzers.targets

```diff
@@ -64,6 +64,9 @@ Copyright (c) .NET Foundation. All rights reserved.
 
     <!-- Enable the trim analyzer when any trimming settings are enabled. Warnings may suppressed based on other settings. -->
     <EnableTrimAnalyzer Condition="'$(EnableTrimAnalyzer)' == '' And ('$(PublishTrimmed)' == 'true' Or '$(IsTrimmable)' == 'true')">true</EnableTrimAnalyzer>
+    
+    <!-- Enable the AOT analyzer when any AOT settings are enabled. Warnings may supressed based on other settings. -->
+    <EnableAotAnalyzer Condition="'$(EnableAotAnalyzer)' == '' And '$(PublishAot)' == 'true'">true</EnableAotAnalyzer>
```

Comment:
I think maybe we should avoid introducing `PublishAot` until it actually does something, otherwise it could be confusing if you set it and the output isn't AOT.

Maybe just add `EnableAotAnalyzer` in this PR?